src/views/product-tutorials-view/helpers/page-data/enrich-learn-data.ts

```diff
@@ -0,0 +1,111 @@
+import {
+	ProductPageBlock as LearnClientProductPageBlock,
+	ProductPageBlockBrandedCallout,
+	ProductPageBlockCardList,
+	ProductPageBlockCollectionsStack,
+	ProductPageBlockFeaturedStack,
+	ProductPageBlockLogoCard,
+	ProductPageBlockTutorialsStack,
+} from 'lib/learn-client/types'
+import { InlineCollections, InlineTutorials } from '../get-inline-content'
+import { formatCollectionCard } from 'components/collection-card/helpers'
+import { CollectionCardPropsWithId } from 'components/collection-card'
+import { TutorialCardPropsWithId } from 'components/tutorial-card'
+import { formatTutorialCard } from 'components/tutorial-card/helpers'
+
+/**
+ * Enriched product page blocks contain specific tutorial & collection data,
+ * rather than slugs. Data is specifically formatted for the props needs
+ * of each block component.
+ */
+export type LearnClientProductPageBlockEnriched =
+	| FeaturedStackEnriched
+	| ProductPageBlockBrandedCallout
+	| CardListEnriched
+	| TutorialsStackEnriched
+	| CollectionsStackEnriched
+// Individual types
+type CardListEnriched = Omit<ProductPageBlockCardList, 'items'> & {
+	collectionCards: CollectionCardPropsWithId[]
+}
+type FeaturedStackEnriched = Omit<ProductPageBlockFeaturedStack, 'blocks'> & {
+	blocks: CardListEnriched[]
+}
+type TutorialsStackEnriched = Omit<
+	ProductPageBlockTutorialsStack,
+	'tutorialSlugs'
+> & {
+	tutorialCards: TutorialCardPropsWithId[]
+}
+type CollectionsStackEnriched = Omit<
+	ProductPageBlockCollectionsStack,
+	'collectionSlugs'
+> & {
+	collectionCards: CollectionCardPropsWithId[]
+}
+
+/**
+ * Format any referenced collection and tutorial slugs,
+ * adding in the full complete data using provided
+ * inlineCollection and inlineTutorial objects.
+ */
+export function enrichLearnData(
```

Comment:
This is the main event of this PR. We take in the authored blocks, as well as the `inlineCollections` and `inlineTutorials`, and format the data so it's ready for the block components to render.

This way we don't have to pass the full `inlineCollections` and `inlineTutorials` to the client, saving a bunch of data (as those would otherwise contain eg full `TutorialLite` listings, which aren't needed for most if not all `block` components).